docs/ARCHITECTURE.rst

```diff
@@ -0,0 +1,132 @@
+=======================
+IDR Client Architecture
+=======================
+
+This document describes among other things the core concepts used through out
+this project, the layout of the project and the class hierarchy of the main
+domain models in the project. All these are important in order to understand
+how the application is structured and how the different components of the
+application fit together and interact with each other.
+
+
+Concepts and Terminology
+------------------------
+
+This section contains important terms and concept central to the project.
+
+Core Domain
+~~~~~~~~~~~
+
+The core domain of the project is majorly composed of the following components:
+
+* **Data Source Type** - A data source type is just that, it describes a kind
```

Comment:
is it possible to name with roman numbers or letters:
a) Data Source or i) Data Source

Reply:
Also, hii part ya data source is just that is not important.

A data source describes the source of data as well as the operations.........